<div style="border:solid orange 2px; padding: 20px"> <h7 style="color:black; margin-bottom:20px">Привет! Меня зовут Егошин Михаил, я буду проводить ревью на этом спринте. Если ты не против, давай общаться на "ты", так я думаю будет комфортнее и удобнее)
Мои комментарии и замечания далее по тексту помечены различными цветами:
    
---    
<div class="alert alert-block alert-danger">
Критическое замечание, которые следует исправить.
</div> 
    
---
    
<div class="alert alert-block alert-warning">
Замечание\совет на будущее.
</div>
    
---
    
<div class="alert alert-block alert-success">
Когда всё сделано правильно.
</div>
    
---

Для меня главное, не просто указать на ошибки, а чтобы у тебя, в первую очередь, было понимание как это делается, подсказать в каких то моментах. Поэтому если возникнут вопросы по проекту или вопросы общего характера, то не стесняйся - задавай!) Для того, чтобы мне было легче их найти, то выделяй их пожалуйста с помощью цвета, например вот так - <font color=orange>"вопрос, который не останется незамеченным"</font>
    
Удачи!
    
p.s.: не удаляй мои замечания, если предстоит что то доработать в проекте.</h7>

<div class="alert alert-block alert-danger">
<b>Комментарии от ревьюера:</b>

Итог ревью - ты проделал большую работу и по большей части (в плане подхода, структуры) успешно. Есть одно критическое замечание, которые тебе следует учесть - выбор корректного актуального периода. Если возникнут вопросы, то пиши здесь - обязательно отвечу. Буду ждать!
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Привет!) Очень рад знакомству)

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

Привет! Все замечания учтены. Успехов на дальнейших спринтах!
</div>

# Введение
Приведен датасет, в котором содержится различная информация об играх с 1980 по 2016 годы, необходимо:
1. Изучить данные

2. Дополнить пропущенную информацию, где это возможно

3. Составить портерт для каждого региона

4. Проверить гипотезы

> 4.1 Средние пользовательские рейтинги платформ Xbox One и PC одинаковые.

> 4.2 Средние пользовательские рейтинги жанров Action и Sports разные.

6. Выявить потенциальное успешные проекты, чтобы спланировать рекламные компании на 2017 год

## Оглавление:
* [Шаг 1.](#1.-Предобработка-данных) Предобработка данных
* [Шаг 2.](#2.-Исследовательский-анализ-данных) Исследовательский анализ данных
* [Шаг 3.](#3.-Портрет-пользователя-каждого-региона) Портреты регионов
* [Шаг 4.](#4.-Проверка-гипотез) Проверка гипотез
* [Вывод](#Итог)

## Описание данных:

* *Name* — название игры
* *Platform* — платформа
* *Year_of_Release* — год выпуска
* *Genre* — жанр игры
* *NA_sales* — продажи в Северной Америке (миллионы проданных копий)
* *E0U_sales* — продажи в Европе (миллионы проданных копий)
* *JP_sales* — продажи в Японии (миллионы проданных копий)
* *Other_sales* — продажи в других странах (миллионы проданных копий)
* *Critic_Score* — оценка критиков (максимум 100)
* *User_Score* — оценка пользователей (максимум 10)
* *Rating* — рейтинг от организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

In [1]:
import pandas as pd
import seaborn as sns
from matplotlib import pyplot as plt
from scipy import stats as st
import plotly.express as px
import numpy as np
pd.options.mode.chained_assignment = None

## 1. Предобработка данных

In [2]:
games =  pd.read_csv('/datasets/games.csv')

In [3]:
games.head(20)

Name Platform  Year_of_Release  \
0                                     Wii Sports      Wii           2006.0   
1                              Super Mario Bros.      NES           1985.0   
2                                 Mario Kart Wii      Wii           2008.0   
3                              Wii Sports Resort      Wii           2009.0   
4                       Pokemon Red/Pokemon Blue       GB           1996.0   
5                                         Tetris       GB           1989.0   
6                          New Super Mario Bros.       DS           2006.0   
7                                       Wii Play      Wii           2006.0   
8                      New Super Mario Bros. Wii      Wii           2009.0   
9                                      Duck Hunt      NES           1984.0   
10                                    Nintendogs       DS           2005.0   
11                                 Mario Kart DS       DS           2005.0   
12                   Pokemon Gold/Pokemon Silver       GB           1999.0   
13                                       Wii Fit      Wii           2007.0   
14                            Kinect Adventures!     X360           2010.0   
15                                  Wii Fit Plus      Wii           2009.0   
16                            Grand Theft Auto V      PS3           2013.0   
17                 Grand Theft Auto: San Andreas      PS2           2004.0   
18                             Super Mario World     SNES           1990.0   
19  Brain Age: Train Your Brain in Minutes a Day       DS           2005.0   

           Genre  NA_sales  EU_sales  JP_sales  Other_sales  Critic_Score  \
0         Sports     41.36     28.96      3.77         8.45          76.0   
1       Platform     29.08      3.58      6.81         0.77           NaN   
2         Racing     15.68     12.76      3.79         3.29          82.0   
3         Sports     15.61     10.93      3.28         2.95          80.0   
4   Role-Playing     11.27      8.89     10.22         1.00           NaN   
5         Puzzle     23.20      2.26      4.22         0.58           NaN   
6       Platform     11.28      9.14      6.50         2.88          89.0   
7           Misc     13.96      9.18      2.93         2.84          58.0   
8       Platform     14.44      6.94      4.70         2.24          87.0   
9        Shooter     26.93      0.63      0.28         0.47           NaN   
10    Simulation      9.05     10.95      1.93         2.74           NaN   
11        Racing      9.71      7.47      4.13         1.90          91.0   
12  Role-Playing      9.00      6.18      7.20         0.71           NaN   
13        Sports      8.92      8.03      3.60         2.15          80.0   
14          Misc     15.00      4.89      0.24         1.69          61.0   
15        Sports      9.01      8.49      2.53         1.77          80.0   
16        Action      7.02      9.09      0.98         3.96          97.0   
17        Action      9.43      0.40      0.41        10.57          95.0   
18      Platform     12.78      3.75      3.54         0.55           NaN   
19          Misc      4.74      9.20      4.16         2.04          77.0   

   User_Score Rating  
0           8      E  
1         NaN    NaN  
2         8.3      E  
3           8      E  
4         NaN    NaN  
5         NaN    NaN  
6         8.5      E  
7         6.6      E  
8         8.4      E  
9         NaN    NaN  
10        NaN    NaN  
11        8.6      E  
12        NaN    NaN  
13        7.7      E  
14        6.3      E  
15        7.4      E  
16        8.2      M  
17          9      M  
18        NaN    NaN  
19        7.9      E

In [4]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
Name               16713 non-null object
Platform           16715 non-null object
Year_of_Release    16446 non-null float64
Genre              16713 non-null object
NA_sales           16715 non-null float64
EU_sales           16715 non-null float64
JP_sales           16715 non-null float64
Other_sales        16715 non-null float64
Critic_Score       8137 non-null float64
User_Score         10014 non-null object
Rating             9949 non-null object
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


In [5]:
#Генерируем словарь в котором каждому названию столбца соответсвует его имя в нижнем регистре
new_names = {}
for column_name in games.columns:
    new_names[column_name] = column_name.lower()

In [6]:
games = games.rename(columns=new_names, inplace=False)

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

Здесь всё хорошо. Ты корректно выгрузил данные и посмотрел их содержимое. Выполнил первичную проверку на пропуски
</div>

### Работа с пропусками и типами

In [7]:
games.isna().sum()

name                  2
platform              0
year_of_release     269
genre                 2
na_sales              0
eu_sales              0
jp_sales              0
other_sales           0
critic_score       8578
user_score         6701
rating             6766
dtype: int64

In [8]:
#Приводим год выхода к int, заменяя NA на 0.
#Пропуски в этом столбце невозможно заполнить, т.к нельзя взять мцд и не невозможно установить точный год автоматически


#games['year_of_release'] = games['year_of_release'].fillna(0).astype('int')

games = games.dropna(subset=['year_of_release'])

<div class="alert alert-block alert-danger">
<b>Комментарии от ревьюера:</b>

А что нам это даст? Тем более когда не можем установить точный год. Корректнее было бы посмотреть объем таких данных и принять решение об удалении
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Пересмотрел еще раз, действительно, большой пользы в этом действии нет. Ведь данные у которых год был заменен на 0 и так не попадут в выборку по актуальному периоду. Более того, удаление 269 элементов из 16700 строк - не очень большая потеря. Решение изменил, старое закоментировал. 

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

Всё верно. Теперь корректно
</div>

In [9]:
#Удалим игры без названия, скорее всего это старые аркадные игры, для которых нельзя однозначно определить имя.
games = games.dropna(subset={'name'})

<div class="alert alert-block alert-warning">
<b>Комментарии от ревьюера:</b>

Здесь ты смело удаляешь (и это правильное решение), хотя опять же не посмотрел объем подобных данных
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> В начале пункта "Работа с пропусками и типами" ввел наглядный пересчет пропусков, до этого смотрел их в первом пункте по количеству записей через .info()

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

Да, это хороший ход)
</div>

In [10]:
#Заполним пропуски в Рейтинге(rating) на основании предположения о том, что игры одного жанра имеют схожую возрастную категорию.
#Логично, что шутеры чаще имеют высокий возрастной рейтинг чем спорт или гонки.

#generes_mode_rating = games.pivot_table(index='genre', values='rating', aggfunc=lambda x: x.mode().iat[0])

<div class="alert alert-block alert-danger">
<b>Комментарии от ревьюера:</b>

А вот здесь ты встаешь на очень опасную дорогу. Заменять данные в реальных задачах очень опасное дело, т.к. это может повлиять на результаты исследования! Первым делом узнают откуда появились пропуски (технический сбой или нет данных) и устраняют проблему. В нашей же задаче есть этап оценки корреляции этих показателей, поэтому замена на  значение моды будет некооректным.
    
Могу отметить, что замена подобным образом, сгруппированное по еще дополнительным признакам (жанр, регион и т.д.) - хороший прием, который может пригодится в дальнейшем опыте. Но в рамках данной задачи этот прием не совсем подходит
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Спасибо за развернутый ответ. Сначала подумал что этот рейтинг можно считать MCAR, что вроде бы дает право их заменить, но теперь понимаю, что для определения типа пропусков нужно копать глубже. Старое решение закоментировал.

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

Отлично!
</div>

In [11]:
def fill_rating_by_genre(row):
    
    game_genre = row['genre']
    game_rating = row['rating']
    
    if pd.isnull(game_rating):
        return generes_mode_rating.loc[game_genre]['rating'][0]
    else:
        return game_rating

In [12]:
games['rating'] = games.apply(fill_rating_by_genre, axis=1)

NameError: ("name 'generes_mode_rating' is not defined", 'occurred at index 1')

In [ ]:
games['user_score'] = pd.to_numeric(games['user_score'], errors='coerce')

### Дополнительные рассчеты

In [ ]:
#Рассчитаем суммарное значение продаж по всем регионам
games['total_sales'] = games['na_sales'] + games['eu_sales'] + games['jp_sales'] + games['other_sales']

### Вывод

Пропуски заполнены, там, где это возможно. Данные приведены к числовым типам там, где это возможно.

## 2. Исследовательский анализ данных

### Исследуем распределение игр по годам

In [ ]:
sns.distplot(games.query('year_of_release > 0')['year_of_release'], bins=40)
plt.title('Количество выпущенных игр по годам')
plt.xlabel('Год выпуска')
plt.ylabel('Частота')
plt.show()

На графике видно, что с течение времени выходит все больше и больше игр. Это связано с постоянным ростом отрасли. Однако, к 2015 году количество выпускаемых игр значительно снизилось. 

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

Не могу не отметить, что ты не просто продемонстрировал умение строить графики, но и не забываешь про оформление (есть заголовок и оси подписаны). Так гораздо информативнее и наглядно без контекста. Молодец!


</div>

In [ ]:
#Выделим топ-10 платформ по общим продажам
top10_platofrm_sales = (games.pivot_table(
    index='platform', values=['total_sales'], aggfunc='sum').sort_values(by='total_sales', ascending=False).head(10))

In [ ]:
top10_platofrm_sales_dict = top10_platofrm_sales.reset_index()['platform'].to_list()

In [ ]:
top10_platofrm = games.query('platform in @top10_platofrm_sales_dict')
top10_platofrm

In [ ]:
pivot_top_10 = top10_platofrm.pivot_table(index=['platform', 'year_of_release'],
                                          values='name', 
                                          aggfunc='count',
                                         ).reset_index()

In [ ]:
for platform in top10_platofrm_sales_dict:  
    sns.lineplot(
        data=pivot_top_10
        .query('platform == @platform and year_of_release !=0 '),
        x='year_of_release',
        y='name',
        label = platform 
    )
    
    plt.gcf().set_size_inches(16, 8)
plt.title('Количество выпущенных игр на платформу по годам')
plt.xlabel('Год выпуска')
plt.ylabel('Количество игр')
plt.show()

На графике видно, что среднее время жизни платформы составляет около 5 лет. Актуальный период будет начинаться с 2008 года,  чтобы отсечь ретро-консоли. 

<div class="alert alert-block alert-danger">
<b>Комментарии от ревьюера:</b>

Если честно на самом графике мне не видно, что среднее время жизни платформы около 5 лет. И по поводу актуального периода я не согласен. Какой смысл рассматривать данные за 9 лет до прогноза на 2017 год, когда к тому моменту многие консоли уже будут морально устарвешими?
    

</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Сейчас постарюсь обьяснить, как я мыслил, и где скорее всего допустил ошибку) Я подумал, что популярность предыдущих генераций консолей xbox, ps и ds опредленно связано с популярностью следующих генераций. Поэтому я отсекал +- по моменту выхода xbox 360, ps3 и ds, чтобы учесть их при анализе.
    
Ниже я изменил актуальный период, теперь он начинается c <b>2014 г.</b>
    
Это так же повлекло за собой изменения в портетах регионов и финальном выводе. Их я так же изменю)

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

Спасибо за разъяснение. Некоторая связь действительно есть. На примере той же PS3/PS4, но в рамках задачи нам эта связь не даст полезной информации
</div>

In [ ]:
games_actual = games.query('year_of_release >= 2014')

In [ ]:
year_sales_actual = games_actual.pivot_table(
    index=['platform','year_of_release'],
    values='total_sales',
    aggfunc = 'sum'
)

In [ ]:
platform_sales = games_actual.pivot_table(
    index='platform', 
    values='total_sales', 
    aggfunc='sum').sort_values(by ='total_sales',ascending=False)

In [ ]:
fig = px.line(
    year_sales_actual.reset_index(),
    x='year_of_release',
    y='total_sales',
    color='platform',
    labels={
        'year_of_release':'Год',
        'total_sales':'Продажи за год'
    },
    title='Количество проданных игр по годам'
    
)

fig.show()

На графике видно, что наибольшие продажи имеют две платформы - PS4 и XBOX One. Обе этих платформы являются новыми поколениями уже существующих и популярных ранее платформ PS3 и XBOX 360

In [ ]:
unique_platforms = year_sales_actual.reset_index()['platform'].unique()

In [ ]:
for platform in unique_platforms:
    data = games_actual.query('platform == @platform')['total_sales']
    sns.boxplot(data)
    plt.title(platform)
    plt.show()

Ситуация на боксплотах очень похожа, сильное искажение вносят игры-хиты, которые продаются огромным тиражом.

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

Здорово, что используешь такой эффективный инстурмент как boxplot
    

</div>

### Оценим корреляцию между оценками игры и ее продажами

In [ ]:
sales_scores_ps4 = games_actual.dropna(subset=['user_score', 'critic_score']).query(' platform == "PS4" ')

In [ ]:
sales_scores_ps4['user_score'] = pd.to_numeric(sales_scores_ps4['user_score'], errors='coerce')

In [ ]:
sales_scores_ps4['critic_score'] = pd.to_numeric(sales_scores_ps4['critic_score'], errors='coerce')

In [ ]:
sns.scatterplot(data = sales_scores_ps4.sort_values(by='total_sales'),
                x='user_score', 
                y='total_sales')
plt.gcf().set_size_inches(16, 8)

plt.xlabel('Оценка пользователей')
plt.ylabel('Мировые продажи ')
plt.gcf().set_size_inches(16, 8)
plt.title('График корреляции')
plt.show()

In [ ]:
sns.scatterplot(data = sales_scores_ps4.sort_values(by='total_sales'),
                x='critic_score', 
                y='total_sales'
               )
plt.xlabel('Оценка критиков')
plt.ylabel('Мировые продажи ')
plt.gcf().set_size_inches(16, 8)
plt.title('График корреляции')
plt.show()

In [ ]:
sales_scores_ps4[['total_sales','user_score','critic_score']].corr(method='spearman')

Из графиков и рассчета коэффициента корреляции 

In [ ]:
genres_count_sale = games_actual.pivot_table(
    index='genre',
    values=['name','total_sales'],
    aggfunc={'name':'count',
             'total_sales':'sum'
            }
)

In [ ]:
genres_count_sale = genres_count_sale.reset_index()

In [ ]:
sns.barplot(data=genres_count_sale, x='genre', y='name')
plt.gcf().set_size_inches(16, 8)
plt.xlabel('Количество выпущенных игр')
plt.ylabel('Жанр ')
plt.title('Количество выпущенных игр по жанрам ')
plt.show()

In [ ]:
sns.barplot(data=genres_count_sale, x='genre', y='total_sales')
plt.gcf().set_size_inches(16, 8)
plt.xlabel('Количество проданных игр')
plt.ylabel('Жанр ')
plt.title('Количество продаж игр по жанрам ')
plt.show()

### Вывод

Графики корреляции показывают, что продажи игры больше зависят от оценок критиков, чем от оценок пользователей. 

На графиках видно, что Action является доминирующим жанром не только по количеству выпускаемых игр, но и по продажам. Adventure имеет достаточно низкие продажи по сравненение с количеством выпускаемых игр. Жанр Shooter имеет высокие продажи, но количество выпускаемых игр сравнительно не велико.



<div class="alert alert-block alert-warning">
<b>Комментарии от ревьюера:</b>

   
В целом, анализ проведен верно. 
    
Стоит отметить еще, что данная корреляция не говорит о причинно-следственной связи. По диаграммам рассеяния видим, что чем выше оценки критиков, тем выше продажи игр на этих платформах, но мы не можем утверждать, что высокие оценки критиками приводят к высоким продажам игр.
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> Возможно я плохо выразился, чем ввел в заблуждение. Я не хотел сказать, что утверждаю наличие связи, просто скаттер с оценками критиков больше походил на корреляцию, чем скаттер с оценками пользователей. 

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

С этим я не спорю) всё верно
</div>

## 3. Портрет пользователя каждого региона

Чтобы не проводить однотипные действия для каждого региона, напишем функции.

In [ ]:
region_rus_names = {
    'na_sales':'Северная америка',
    'eu_sales':'Европа',
    'jp_sales':'Япония',
    'other_sales':'Другие регионы'
}

In [ ]:
def find_top5_platforms(df, region):
    
    print(f'Топ-5 платформ в регионе {region_rus_names[region]}')
    
    #Сводная таблица по региону.
    pivot_platform_region = df.pivot_table(
        index='platform',
        values=region,
        aggfunc='sum'
    )
    
    #Построение графика 
    pivot_platform_region.sort_values(by=region, ascending=False).head(5).plot.pie(y=region,
                                                                                   autopct='%1.1f%%')
    plt.title('Распредление долей продаж по платформам')
    plt.gcf().set_size_inches(12, 10)
    
    return pivot_platform_region.sort_values(by=region, ascending=False).head(5)
    

In [ ]:
def find_top5_genres(df, region):
    
    print(f'Топ-5 жанров в регионе {region_rus_names[region]}')
    
    #Сводная таблица по региону.
    pivot_genre_region = df.pivot_table(
        index='genre',
        values=region,
        aggfunc='sum'
    )
    
    #Построение графика 
    pivot_genre_region.sort_values(by=region, ascending=False).head(5).plot.pie(y=region,
                                                                                autopct='%1.1f%%')
    plt.title('Распредление долей продаж по жанрам')

    plt.gcf().set_size_inches(12, 10)
    
    return pivot_genre_region.sort_values(by=region, ascending=False).head(5)

In [ ]:
def find_corr_ESRB_sales(df, region):
    
    sns.barplot(
        data = df,
        y = region,
        x = 'rating'
    )
    
    plt.title('Влияние возрастного рейтинга на продажи')
    plt.gcf().set_size_inches(8, 6)

In [ ]:
def find_dependence_ESRB_sales(df, region):
    
    pivot = df.pivot_table(
        index='rating',
        values=region,
        aggfunc='sum'
    ).sort_values(by=region, ascending=False).reset_index()
    
    ax = sns.barplot(data=pivot, y=region, x='rating')
    ax.set(xlabel='Количество продаж', ylabel='Возрастной рейтинг')
    plt.title('Соотнешение продаж и возрастного рейтинга')
    plt.show()

### Портерт региона Северная Америка

In [ ]:
find_top5_platforms(games_actual, 'na_sales')

In [ ]:
find_top5_genres(games_actual, 'na_sales')

In [ ]:
find_dependence_ESRB_sales(games_actual, 'na_sales')

### Описание региона 
В Cеверной Америке предпочитают Action, Shooter, Sports а играют в основном на платформах PS4 и XBOX One

### Портерт региона Европа

In [ ]:
find_top5_platforms(games_actual, 'eu_sales')

In [ ]:
find_top5_genres(games_actual, 'eu_sales')

In [ ]:
find_dependence_ESRB_sales(games_actual, 'eu_sales')

### Описание региона 
В Европе предпочитают Ps4, она занимает более половины рынка самые популярные жанры это Action, Shooter и Sports.  

### Портерт региона Япония

In [ ]:
find_top5_platforms(games_actual, 'jp_sales')

In [ ]:
find_top5_genres(games_actual, 'jp_sales')

In [ ]:
find_dependence_ESRB_sales(games_actual, 'jp_sales')

### Описание региона
В японии наиболее популярны платформы местного производства: 3DS, PS4. Доминирующим жанрами являются role-play и Action. 

### Портерт других Регионов

In [ ]:
find_top5_platforms(games_actual, 'other_sales')

In [ ]:
find_top5_genres(games_actual, 'other_sales')

In [ ]:
find_dependence_ESRB_sales(games_actual, 'other_sales')

### Описание региона 
В других регионах предпочитают Action и Shooter, доминирующей платформой является PS4 (почти 60% рынка) 

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

К этому этапу замечаний нет. Удачные выбор графика (наглядно), выводы соответствуют полученным результатам
</div>

## 4. Проверка гипотез

Проверяем гипотезу о том, средние пользовательские рейтинги платформ Xbox One и PC одинаковые

H0:Средние двух генеральных совокупностей равный между собой

H1:Средние двух генеральный совокупностей не равный между собой

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

Очень здорово, что ты не забыл про формулировки гипотез. Это важный пункт в подобного рода задачах
</div>

In [ ]:
sample_XOne = games_actual.query('platform == "XOne" ')['user_score'].dropna()

In [ ]:
sample_PC = games_actual.query('platform == "PC" ')['user_score'].dropna()

In [ ]:
sample_PC.describe()

In [ ]:
sample_XOne.describe()

In [ ]:
alpha = 0.05

In [ ]:
results = st.ttest_ind(sample_XOne, sample_PC, equal_var=False)
results.pvalue

In [ ]:
if (results.pvalue < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось отвергнуть нулевую гипотезу")

Проверяем гипотезу о том, что средние пользовательские рейтинги жанров Action (англ. «действие», экшен-игры) и Sports (англ. «спортивные соревнования») разные.

H0:Средние двух генеральных совокупностей равный между собой

H1:Средние двух генеральный совокупностей не равный между собой

In [ ]:
sample_action =  games_actual.query('genre == "Action" ')['user_score'].dropna()

In [ ]:
sample_action.describe()

In [ ]:
sample_sports =  games_actual.query('genre == "Sports" ')['user_score'].dropna()

In [ ]:
sample_sports.describe()

In [ ]:
results = st.ttest_ind(sample_action, sample_sports, equal_var=False)
results.pvalue

In [ ]:
if (results.pvalue < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось отвергнуть нулевую гипотезу")

### Вывод

Результат проверки первой гипотезы говорит нам о том, что нет оснований пологать, что средние пользовательские рейтинги игр на Xbox One и PC различаются.

Результат проверки второй гипотезы говорит нам о том, что средние пользовательские рейтинги жанров Action и Sports скорее всего различаются. 

Для проверки обеих гипотез нам необходимо сравнить средние двух случайно распределенных величин, это возможно сделать при помощи критерия Стьюдента. Нулевая и альтернативные гипотезы выдвигаются идентично друг другу. H0 - средние значение равны между собой и H1 - средние значение не равны между собой. 

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

Корректный результат. Очень хорошо, что используешь параметр equal_var (с ним результат точнее).
</div>

# Итог

Основной задачей проекта является выбор маркетинговых компаний для магазина на 2017год.
Распределение платформ и жанров показывает, что есть смысл организовать две раздельные рекламные компании, одну для Японии, а другую для всех остальных регионов. 

Так например в Японии более полупулярны консоли локального производства, как например 3DS и DS, а так же жанр Role Playing, являющийся доминирующим. Это означает, что в Японии стоит потратить деньги на рекламу этого жанра для этих консолей. При этом стоит обратить внимание на возрастной рейтинг игры - для Японии наибольшие продажи имеют рейтинги T и E.

Для всего остального мира стоит рекламировать жанры Shooter и Action, на платформы XBOX 360, PS3 а так же на следующие генерации этих консолей. X360 и PS3 лидируют по продажам в длительный период времени и множество людей имеют эти консоли, а PS4 и XONE набирают обороты к 2017 году. По рейтингам предпочтительными будут E и М.

Более того, портреты регионов (кроме японии) показывают популярность двух платформ: XBOX и PS, одни и те же игры выходят на эти платформы паралельно, что позволяет экономить деньги на производстве маркетинговых материалов. 

Так же, в ходе работы было выявлено, что оценки критиков влияют на продажи игры сильнее, чем пользовательские рейтинги. Оценку критиков игры получают в первые дни релиза, а может быть и за некоторое время до него, это означает, что в рекламных компаниях следует продвигать проекты, получившие высокие оценки экспертов индустрии. 

Результат проверки первой гипотезы говорит нам о том, что нет оснований пологать, что средние пользовательские рейтинги игр на Xbox One и PC различаются.

Результат проверки второй гипотезы говорит нам о том, что средние пользовательские рейтинги жанров Action и Sports скорее всего различаются. 


<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера:</b>

Итоговый вывод завершает твое исследование. Самое главное - ты дал ценные рекомендации бизнесу, а это очень важное качество  в работе аналитика. Развивай его и дальше!
</div>

<div class="alert alert-info"> <b>Комментарий студента:</b> После изменения актуального периода и получения новой информации можно сказать, что рекомендация старых генераций PS и XBOX был неверной и стоит делать упор на XONE и PS4, с большим уклоном в PS4, т.к она доминирует практический на всех рынках и даже нашла свое место в японии. 

<div class="alert alert-block alert-success">
<b>Комментарии от ревьюера v2:</b>

Тут спорить глупо. Как показало время, действительно PS4 доминировала всё это время
</div>